src/routes/github/create-branch/github-create-branch-get.ts

```diff
@@ -77,6 +78,21 @@ export const GithubCreateBranchGet = async (req: Request, res: Response, next: N
 	});
 };
 
+export const generateBranchName = (issueKey: string, issueSummary: string) => {
+	if (!issueSummary) {
+		return issueKey;
+	}
+
+	let branchSuffix = issueSummary;
+	const validBranchCharactersRegex = /[^a-z0-9.\-_]/gi;
```

Comment:
Every special character except for `.` , `_` and `-` gets replaced by a `-`. So if we have multiple special characters at the end, we're gonna end up with multiple `-` at the end, which may look weird for a branch name.
`generateBranchName("ISSUEKEY", "/first-task-by-me-:)")` will return `ISSUEKEY-first-task-by-me--`

Reply:
added a function to clean it up so only one - at a time, also added a test.
Did it manually because our node version doesn't support the new replaceAll function and didn't want to pull on that thread and start updating things as part of this PR